### PR TITLE
New features

### PR DESCRIPTION
**Pull request by anmentone**
Hi, I propose you an implementation for dependency relationship.
Let me know if it's good for the merge.

Good bye!

### CODE REVIEW

File: src/hpp2plantuml/hpp2plantuml.py [hunk 1]
```diff
@@ -1033,12 +1101,12 @@ def build_aggregation_list(self):
                 comp_parent_obj = class_list_obj[comp_parent_idx]['obj']
                 self._aggregation_list.append(
                     ClassAggregationRelationship(
-                        obj_class_obj, comp_parent_obj, comp_count,
+                        comp_parent_obj, obj_class_obj, comp_count,
```
Comment (thibaultmarin) [inline src/hpp2plantuml/hpp2plantuml.py]: I would suggest leaving this change out of this pull request.

I believe the current version (v0.5) is producing the correct output:

    class Engine {};
    class Car{ Engine* engine; };

produces `Car o-- Engine` as expected.

However, the current code is misleading/wrong, thanks for pointing that out.  I
am passing `obj_class_obj` as `c_parent` and `comp_parent_obj` as `c_child` when
instantiating the `ClassAggregationRelationship` object.  This is wrong and your
change is correct, but the code needs to be further modified to produce the same
output as v0.5.  I would propose removing this change and fix that in another
commit (I will take care of that before releasing v0.6).

File: src/hpp2plantuml/hpp2plantuml.py [hunk 2]
```diff
@@ -1033,12 +1101,12 @@ def build_aggregation_list(self):
...
                         flag_use_namespace=flag_use_namespace))
 
     def _augment_comp(self, c_dict, c_parent, c_child):
         """Increment the aggregation reference count
-
+aggregation
```
Comment (thibaultmarin) [inline src/hpp2plantuml/hpp2plantuml.py]: This line looks out of place, maybe a typo?

File: src/hpp2plantuml/templates/default.puml [hunk 3]
```diff
@@ -24,4 +24,11 @@
 {% endfor %}
 {% endblock %}
 
+{% block dependence %}
```
Comment (thibaultmarin) [inline src/hpp2plantuml/templates/default.puml]: dependency?

File: src/hpp2plantuml/hpp2plantuml.py [hunk 4]
```diff
@@ -988,6 +1011,51 @@ def build_inheritance_list(self):
...
+                                # Check if the method param type is a Class type
+                                index = objects_name.index(method[0])
+                            except:
+                                pass
+
+                            if index != ValueError and method[0] != obj.get_name():
+                                depend_obj = self._objects[index]
+
+                                self._dependency_list.append(ClassDependencyRelationship(depend_obj, obj, flag_use_namespace=flag_use_namespace))
+            print()
```
Comment (thibaultmarin) [inline src/hpp2plantuml/hpp2plantuml.py]: Can we stay under 79 characters per line?

```suggestion
                            index = ValueError
                            try:
                                # Check if the method param type is a Class
                                # type
                                index = objects_name.index(method[0])
                            except:
                                pass

                            if index != ValueError and \
                               method[0] != obj.get_name():
                                depend_obj = self._objects[index]

                                self._dependency_list.append(
                                    ClassDependencyRelationship(
                                        depend_obj, obj,
                                        flag_use_namespace=flag_use_namespace))
```

File: src/hpp2plantuml/hpp2plantuml.py [hunk 5]
```diff
@@ -664,6 +667,7 @@ def _render_link_type(self):
         str
             The link between parent and child following the PlantUML syntax
         """
+        # print(self._link_type)
```
Comment (thibaultmarin) [inline src/hpp2plantuml/hpp2plantuml.py]: We could delete debugging code I think

File: src/hpp2plantuml/hpp2plantuml.py [hunk 6]
```diff
@@ -648,6 +650,7 @@ def render(self):
         # Link string
         link_str += parent_str + ' ' + self._render_link_type() + \
                     ' ' + child_str + '\n'
+        # print(link_str)
```
Comment (thibaultmarin) [inline src/hpp2plantuml/hpp2plantuml.py]: We could delete debugging code I think

**Comment by thibaultmarin** [comment]
Please take a look at v0.6 which integrates your changes with minor changes and let me know if fits your needs.

Thanks

**Comment by anmentone** [comment]
I apologize if I didn't check you answers, I had some problems. I agree with your suggestions and your changes work well for my needs, thank you for your upgrade!

**Comment by thibaultmarin** [comment]
Thanks for your contribution.

**Pull request closed without merging**